sound/soc/intel/boards/bxt_tdf8532.c

```diff
@@ -217,6 +217,19 @@ static struct snd_soc_dai_link broxton_tdf8532_dais[] = {
 		.dpcm_playback = 1,
 		.no_pcm = 1,
 	},
+	{
+		/* SSP2 Dirana for vm1 */
+		.name = "vm1_dai_linkC",
```

Comment:
same here, why would one add a new BE dailink? This is not quite right.

Reply:
No. There should be no additional DAI links. the BE should use a PCM FE and rely on firmware to route PCM samples to the DAI. what you are implementing is hardware partitioning, this is not what we want. The Virtio BE shall not deal directly with DAI links or even know about them.

Reply:
"the BE should use a PCM FE and rely on firmware to route PCM samples to the DAI" works for the native driver without audio mediator support. However audio mediator is a little different. 

In native driver, user opens PCM0. The audio driver knows everything, such as which codec dai it is using (codec dai is bound to the BE dai link and so it is bound to the PCM device). So ASoC will setup the codec and so on.

Please image:
User triggers playback in vFE. vFE notifies vBE we will playback on the PCM0 in vFE (PCM0 device is in vFE. vBE's PCM0 is different from vFE's PCM0. vBE's PCM0 is exposed to the user in SOS). So vBE must route PCM samples to the correct DAI. vBE get the playback trigger from vFE. It is not a trigger from vBE userspace. So ASoC will not set the HW setting automatically. Please notice, HW setting which ASoC sets is triggered by the userspace. However, playback in vFE will not trigger SOS ASoC operating. So vBE must handle the setting itself.

in native:
userspace <--> alsa <---> ASoC <---> cpu dai/codec dai

in audio mediator:
vFE userspace <---> vFE alsa <---> vFE ASoC <---> vFE audio driver <---> vBE audio driver <---> (SOF <---> cpu_dai)/codec dai

Reply:
Sorry, I don't get at all why you would need the vBE to know anything about DAIs. I am not going to continue this discussion and will revert everything related to virtualization in sof-dev since it's a distraction at this point.